### DISCUSSION TITLE
There is a bug with custom payment providers in 2.0

### DISCUSSION THREAD

**Discussion by JesusDR01** [medusajs/medusa#10263] [Q&A]
Hi,

I’m trying to migrate the Klarna module to MedusaJS 2.0, and I believe there might be a bug when creating a new payment provider module in the backend.

Currently, I have my module imported using the new architecture, and the database now contains the new payment provider ID in the table.

However, the backend is returning the error: Could not find a payment provider with id: pp_klarna_klarna when I call the endpoint /store/payment-collections/pay_col_01JDHCVTVF3XAJR05RN0C3ZVV2/payment-sessions, using provider_id: "pp_klarna_klarna"

As far as I can see, the payment provider is enabled, created in the database, and activated in the admin panel.

How can I resolve this issue?

Thank you!

![image](https://github.com/user-attachments/assets/278a1a16-6d62-450a-8e34-f550cdd0fda7)
![image](https://github.com/user-attachments/assets/36acfdcb-2a8e-42cf-b47a-a15e82de2327)
![image](https://github.com/user-attachments/assets/7954aff3-99af-4b3c-bf21-4cd3f3579bfe)
<img width="341" alt="image" src="https://github.com/user-attachments/assets/c94dd78d-ad4a-4c73-a758-811c96a8cd3c">

**Comment by azimsb120** [comment]
any updates? i see a similar problem.

**Comment by icanos** [comment]
Anyone successfully migrated the Klarna payment plugin to v2 architecture? I started looking into building one, but weren't able to figure out how to get the cart items in the payment provider.

**Comment by schnecktec** [comment]
You might be registering the same module resolve: "@medusajs/medusa/payment" twice. You need to list it just once in medusa-config.ts and then add several providers: providers: [{provider1},{provider2}]
